chalk-rust-ir/src/lib.rs

```diff
@@ -103,6 +103,23 @@ pub struct StructFlags {
     pub fundamental: bool,
 }
 
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub struct FnDefDatum<I: Interner> {
+    pub id: FnDefId<I>,
+    pub binders: Binders<FnDefDatumBound<I>>,
+    pub flags: FnDefFlags,
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Hash, Fold, HasInterner)]
+pub struct FnDefDatumBound<I: Interner> {
+    pub argument_types: Vec<Ty<I>>,
+    pub return_type: Ty<I>,
+    pub where_clauses: Vec<QuantifiedWhereClause<I>>,
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub struct FnDefFlags {}
```

Comment:
Hmm an empty set of flags may not be worth including unless we expect to put things in here...